### DISCUSSION TITLE
Embedded language support (Vue, Svelte, Astro ...)

### DISCUSSION THREAD

**Discussion by Hebilicious** [biomejs/biome#1691] [General]
## Motivation

There's a lot of existing discussions that are all around the same topic. It would be great to centralise the discussion somewhere.

Adding support for these would increase the number of users/contributors, as these frameworks have a big combined user base.
oxlint recently added support for these to lint the content of the script tag. I think (as someone who knows nothing about Rust and the Biome codebase) that this can be done "quickly", ie just extracting the JS/TS parts as mentioned by @anonrig

I would assume that these should share some common code, but there would be a need for a dedicated plugin for each of them eventually ?

Possibly relevant for type related rules : https://volarjs.dev/core-concepts/embedded-languages/, can create TS language servers that works with embedded languages (This is what Vue and Astro use for their vscode extensions under the hood).

As someone who uses both Vue and Svelte, I would be happy to contribute with something for these.

## Existing discussions references :

- https://github.com/biomejs/biome/discussions/136
- https://github.com/biomejs/biome/discussions/372
- https://github.com/biomejs/biome/discussions/382
- https://github.com/biomejs/biome/discussions/1254
- https://github.com/biomejs/biome/discussions/1678

**Comment by Netail** [comment]
Hey!
As of [Biome 2.3](https://biomejs.dev/blog/biome-v2-3/) we support HTML, Astro, Svelte & Vue under the experimental option: `html.experimentalFullSupportEnabled`.
Support is still experimental due to potentially missed edge cases, but definitely worth trying :) 
In case you find any issues please make sure to [create a GitHub issue](https://github.com/biomejs/biome/issues/new/choose) so we can get closer to stable :)

More on HTML-ish languages can be found here: https://biomejs.dev/internals/language-support/#html-super-languages-support
